Fixes the space heater right click functionality

HarseTheef opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Currently the space heater does not properly toggle on/off the functionality of the space heater itself, just one aspect of its icon state.  This brings the right click functionality in line with the TGUI on/off toggle functionality.  It will now properly inform the user via chat when it is toggled and also properly change the state of the machinery to be on and off.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
fixes #63042 and standardizes the toggling of the machine.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
fix: The space heater right click toggle now works correctly.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Commit:
Fixes the space heater right click functionality

previously the space heater was not toggling its on/off state, staying consistent with the TGui functionality of turning it on and off.  This corrects such.

JohnFulpWillard commented on code/game/machinery/spaceheater.dm:
```diff
@@ -186,9 +186,11 @@
 		return
 	on = !on
+	mode = HEATER_MODE_STANDBY
+	usr.visible_message(span_notice("[usr] switches [on ? "on" : "off"] \the [src]."), span_notice("You switch [on ? "on" : "off"] \the [src]."))
```
```suggestion
	user.visible_message(span_notice("[user] switches [on ? "on" : "off"] \the [src]."), span_notice("You switch [on ? "on" : "off"] \the [src]."))
```

Why not use user here instead?

HarseTheef replied:
It wasn't a consideration at the time, but I am open to it.  I just tested it locally.  What are the benefits/functionality changes?

Commit:
adjustment to the chat message

changes usr to user

LemonInTheDark commented on code/game/machinery/spaceheater.dm:
```diff
@@ -186,9 +186,11 @@
 	if(!can_interact(user))
 		return
 	on = !on
+	mode = HEATER_MODE_STANDBY
+	user.visible_message(span_notice("[user] switches [on ? "on" : "off"] \the [src]."), span_notice("You switch [on ? "on" : "off"] \the [src]."))
 	update_appearance()
 	if (on)
-		START_PROCESSING(SSmachines, src)
+		SSair.start_processing_machine(src)
```
This is all the same as the logic in ui_act right? could we move it to a separate proc and just call that instead of having two implementations of the same thing?

Would prevent errors like this in the future too

HarseTheef replied:
Yes, it is!  I have moved it to a separate proc which is now called by both the right click action and the tgui functionality.  Good catch!

Commit:
migrates toggling the space heater state to a proc

toggling the space heater on and off is now a proc called by both the ui and the right click functionality

LemonInTheDark commented on code/game/machinery/spaceheater.dm:
```diff
@@ -265,6 +257,14 @@
 	panel_open = TRUE
 	update_appearance()
 
+/obj/machinery/space_heater/proc/toggle_space_heater_state()
```
You could prob just have this be `toggle_on` or something, but it's not a huge deal.

HarseTheef replied:
I actually considered that, but the same proc can also toggle it off, and `toggle_state` is a global.

LemonInTheDark replied:
`toggle_heat`?

HarseTheef replied:
I considered that too, but since that could be confused for changing the mode of the space heater from one of auto/heat/cool I didn't want to create confusion with its current functionality.  Any other ideas?  I would be happy to change it to something more concise that didn't create confusion.

HarseTheef replied:
I changed it to `toggle_power` .  I think that will work without creating confusion.  Thoughts?

LemonInTheDark replied:
Seems good to me

Commit:
makes proc name more concise

changes the proc name to toggle_power
